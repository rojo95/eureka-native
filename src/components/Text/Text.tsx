import { customText } from "react-native-paper";

// Use this instead of importing `Text` from `react-native-paper`
const Text = customText<"customVariant">();

export default Text;
