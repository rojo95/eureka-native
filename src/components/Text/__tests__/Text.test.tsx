import React from "react";
import renderer from "react-test-renderer";

import Text from "../Text";

describe("Testing about <Text/> component ", () => {
    const testText = "Testeo";
    it("render correctly", () => {
        const tree: any = renderer.create(<Text>{testText}</Text>).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it("show correct text", () => {
        const tree: any = renderer.create(<Text>{testText}</Text>).toJSON();
        expect(tree.children[0]).toBe(testText);
    });

    it("render text without change", () => {
        const tree = renderer.create(<Text>{testText}</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
