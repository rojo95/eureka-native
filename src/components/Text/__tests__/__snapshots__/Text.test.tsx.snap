// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing about <Text/> component  render text without change 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      undefined,
    ]
  }
>
  Testeo
</Text>
`;
